src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,215 @@
-        return null;
+        Node<T> node = head;
+        if (index > 0 && index < (size)) {
+            for (int i = 0; i < size; i++) {
+                if (i == index) {
+                    if (node.next != null) {
+                        node.next.prev = node.prev;
+                    }
+                    node.prev.next = node.next;
+                    size--;
+                    return node.value;
+                } else {
+                    node = node.next;
+                }
+            }
+        } else if (index == 0) {
+            if (size != 1) {
+                node.next.prev = null;
+                head = node.next;
+                size--;
+                return node.value;
+            } else {
+                head = null;
+                tail = null;
+                size--;
+                return node.value;
+            }
+
+        }
+        {
+            throw new IndexOutOfBoundsException("size; " + size + "index: " + index);
+        }
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> node = head;
+        if (size == 1) {
+            head = null;
+            size--;
+            return true;
+        } else {
+            for (int i = 0; i < size; i++) {
+                if (node.value == null && object == null) {
+                    node.next.prev = node.prev;
+                    node.prev.next = node.next;
+                    size--;
+                    return true;
+                } else if (node.value != null) {
+                    if (node.value.equals(object)) {
+                        if (node.equals(tail)) {
+                            node.prev.next = null;
+                            tail = node.prev;
+                            size--;
+                            return true;
+                        } else if (node.equals(head)) {
+                            head = node.next;
+                            head.prev = null;
+                            size--;
+                            return true;
+                        } else {
+                            node.prev.next = node.next;
+                            node.next.prev = node.prev;
+                            size--;
+                            return true;
+                        }
+                    }
+                    node = node.next;
+                } else {
+                    node = node.next;
+                }
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        Node(Node<T> prev,T value, Node<T> next) {
+            this.value = value;
+            this.prev = prev;
+            this.next = next;
+        }
     }
```

Comment:
The checklist states that we shouldn't use getters and setters in the `Node` class. In your implementation, you've correctly accessed the private fields of the inner class directly, which is a good practice as it makes the code more concise.